Content.Server/Medical/SuitSensors/SuitSensorSystem.cs

```diff
@@ -0,0 +1,291 @@
+                    name = "suit-sensor-mode-vitals";
+                    break;
+                case SuitSensorMode.SensorCords:
+                    name = "suit-sensor-mode-cords";
+                    break;
+                default:
+                    return "";
+            }
+
+            return Loc.GetString(name);
+        }
+
+        public void SetSensor(EntityUid uid, SuitSensorMode mode, EntityUid? userUid = null,
+            SuitSensorComponent? component = null)
+        {
+            if (!Resolve(uid, ref component))
+                return;
+
+            component.Mode = mode;
+
+            if (userUid != null)
+            {
+                var msg = Loc.GetString("suit-sensor-mode-state", ("mode", GetModeName(mode)));
+                _popupSystem.PopupEntity(msg, uid, Filter.Entities(userUid.Value));
+            }
+        }
+
+        public SuitSensorStatus? GetSensorState(EntityUid uid, SuitSensorComponent? sensor = null, TransformComponent? transform = null)
+        {
+            if (!Resolve(uid, ref sensor, ref transform))
+                return null;
+
+            // check if sensor is enabled and worn by user
+            if (sensor.Mode == SuitSensorMode.SensorOff || sensor.User == null)
+                return null;
+
+            // try to get mobs id from ID slot
+            var userName = Loc.GetString("suit-sensor-component-unknown-name");
+            var userJob = Loc.GetString("suit-sensor-component-unknown-job");
+            if (_idCardSystem.TryGeIdCardSlot(sensor.User.Value, out var card))
+            {
+                if (card.FullName != null)
+                    userName = card.FullName;
+                if (card.JobTitle != null)
+                    userJob = card.JobTitle;
+            }
+
+            // get health mob state
+            if (!EntityManager.TryGetComponent(sensor.User.Value, out MobStateComponent? mobState))
+                return null;
+            var isAlive = mobState.IsAlive();
+
+            // get mob total damage
+            if (!EntityManager.TryGetComponent(sensor.User.Value, out DamageableComponent? damageable))
+                return null;
+            var totalDamage = damageable.TotalDamage.Int();
+
+            // finally, form suit sensor status
+            var status = new SuitSensorStatus(userName, userJob);
+            switch (sensor.Mode)
+            {
+                case SuitSensorMode.SensorBinary:
+                    status.IsAlive = isAlive;
+                    break;
+                case SuitSensorMode.SensorVitals:
+                    status.IsAlive = isAlive;
+                    status.TotalDamage = totalDamage;
+                    break;
+                case SuitSensorMode.SensorCords:
+                    status.IsAlive = isAlive;
+                    status.TotalDamage = totalDamage;
+                    status.Coordinates = transform.MapPosition;
+                    break;
+            }
```

Comment:
```suggestion
            // finally, form suit sensor status
            var status = new SuitSensorStatus(userName, userJob);
            switch (sensor.Mode)
            {
                case SuitSensorMode.SensorCords:
                    status.Coordinates = transform.MapPosition;
                case SuitSensorMode.SensorVitals:
                    // get mob total damage
                    if (EntityManager.TryGetComponent(sensor.User.Value, out DamageableComponent? damageable))
                        status.TotalDamage = damageable.TotalDamage.Int();
                case SuitSensorMode.SensorBinary:
                    // get health mob state
                    if (EntityManager.TryGetComponent(sensor.User.Value, out MobStateComponent? mobState))
                        status.IsAlive = mobState.IsAlive();
                    break;
            }
```
webedited together, check formatting after you commit this one or just code it yourself

Reply:
I've checked in IDE. Code above isn't valid C# code. You can't "fallthrough" case statements in C#. Here is an example which wouldn't compily:
```
private enum X { A, B }

private void Boo()
{
    int i = 0;
    var e = X.A;

    switch (e)
    {
        case X.A:
            i++;
        case X.B:
            i--;
            break;
    }
}
```

Reply:
my bad, you need to write `goto case X;` to create a fallthrough. implicit fallthroughs are disabled in c#, for good reason i might add, so you need to explicitely define them.

that being said, i forgot to mention why i actually wanted this change: i dont like simply a mobstatecomp/damageablecomp not existing killing the entire state from being sent.

Reply:
I've added default values to current code, if components don't exist.